anitab-org/bridge-in-tech-web
# Feat: Update additional info functionality

### Description
Allow useer to update theeir additional information

Fixes #48 

### Type of Change:
- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
- Create 2 users on backend Swagger UI, one with additional information and the other without
- Login as user that has no additional information in the database
- Go to `My Space` > `Personal Details` > `Additional Info`
- see message as below:
<img width="521" alt="Screen Shot 2020-08-03 at 8 05 26 pm" src="https://user-images.githubusercontent.com/29667122/89172212-9ac38580-d5c5-11ea-9111-c6cd2afd8ed2.png">

- Create additional information
- On successful create, see message
<img width="507" alt="Screen Shot 2020-08-03 at 8 05 57 pm" src="https://user-images.githubusercontent.com/29667122/89172270-b333a000-d5c5-11ea-8601-ec0963b4239a.png">

Notice how Timezone is skipped by user (which could happen either accidentally or on purpose if user prefer not to disclose their timezone). If this happens, the system will automatically use the default value of GMT+00 as user timezone. To test this, click away to different page and come back to `My Spacee` > `Additional Info` and see GMT being the value displayed from GET /user/additional_info request.
<img width="479" alt="Screen Shot 2020-08-03 at 8 06 08 pm" src="https://user-images.githubusercontent.com/29667122/89172689-5e445980-d5c6-11ea-97d1-0f2b1098ad7d.png">


- to test Update, now edit the additional information. On successful update, see message below
<img width="524" alt="Screen Shot 2020-08-03 at 8 06 18 pm" src="https://user-images.githubusercontent.com/29667122/89172769-8633bd00-d5c6-11ea-8655-3e84fbb4ac9e.png">

- to check if the data is updated, again, click away to different page and come back to `My Space` > `Additional Info` and see the updated data (in this example, the timezone field)
<img width="492" alt="Screen Shot 2020-08-03 at 8 06 33 pm" src="https://user-images.githubusercontent.com/29667122/89172939-cabf5880-d5c6-11ea-97fa-ca09a158c439.png">



### Checklist:


**Code/Quality Assurance Only**
